code/lora/lora.py

```diff
@@ -0,0 +1,298 @@
+        packet_length = len(self.image_waiting)
+        if packet_length < DOWN_PACKET_LENGTH:
+            self.image_waiting += (DOWN_PACKET_LENGTH-packet_length)*[45]
+        print("TX:", self.image_waiting)
+
+        self.write_payload(HEADER.LORA + self.image_waiting)
+        self.set_mode(MODE.TX)
+        time.sleep(TX_TIME)
+        self.reset_ptr_rx()
+        self.set_mode(MODE.RXCONT)
+        self.last_tx_mode = 'image'
+
+
+    def tx_thermal(self):
+        """
+        Transmits a chunk of MLX90640 thermal image.
+        If no ACK was received, same payload is sent.
+        Otherwise, a new chunk is loaded.
+        Upon transmitting 32 lines, stops the transmission
+        """
+        #NOT TESTED YET
+
+        if self.ack['thermal']:
+            self.thermal_counter += 1
+            if self.thermal_counter > THERMAL_HEIGHT:
+                self.sending_thermal = False
+                return
+            row = [int(i) for i in self.thermal[self.thermal_counter-1]]
+            row[:] = [max(el, 0) if el < 255 else min(el, 255) for el in row]
+            self.thermal_waiting = [HEADER.THERMAL_B] + row
+
+        print(self.thermal_waiting)
+        self.ack['thermal'] = False
+        self.write_payload(HEADER.LORA + self.thermal_waiting)
+        self.set_mode(MODE.TX)
+        time.sleep(TX_TIME)
+        self.reset_ptr_rx()
+        self.set_mode(MODE.RXCONT)
+        self.last_tx_mode = 'thermal'
+
+
+    def wait_for_rx(self):
+        """
+        Waits for slave responce.
+        No sleep() because of expected RX callback
+        """
+
+        start_time = time.time()
+        while time.time() - start_time < RX_TIME:
+            pass
+        self.reset_ptr_rx()
+
+
+    def parse_cmd(self):
+        """
+        Starts / ends transmissions based on received command
+        """
+
+        with open(CONF_FILE, "r") as conf:
+            conf.seek(0, 2)
+            pos = conf.tell()
+            if pos > self.conf_filepos:
+                self.conf_filepos = pos
+                conf.seek(pos - CMD_LENGTH, 0)
+                self.cmd = conf.readline()
+                print("CMD:", self.cmd)
+
+                if self.cmd[0] == "I" and not self.sending_image:
+                    img_path = str(IMAGES) + '/' + str(self.cmd[1:6]) + '_' + IMG_QUAL[self.cmd[6]] + '.jpg'
+                    self.image = open(img_path, "rb")
+                    self.image_size = int(os.path.getsize(img_path))
+                    self.sending_image = True
+                    self.buffer.insert(0, "SI{:07}".format(self.image_size) + self.cmd[1:7] + (DOWN_PACKET_LENGTH - 13)*"-")
+
+                if self.cmd[0] == "T" and not self.sending_thermal:
+                    with open(str(THERMAL)+ '/' + str(self.cmd[1:6])+".txt", "r") as t:
+                        #Sorry, too tempting...
+                        self.thermal = [[int(val) for val in line] for line in [line[:-2].split(" ") for line in t.readlines()]]
```

Comment:
Smartass. How about explaining what it does?

Reply:
The file with image data comprises 32 lines of 24 numbers split by spaces. This comprehension reads the lines, isolates the values from each other and converts them to ints.

Reply:
I'm not sure whether one comprehension is suitable for such process, but alright, let it be